extensions/nginx/index.js

```diff
@@ -25,23 +25,43 @@ class NginxExtension extends Extension {
         }];
     }
 
-    setup(cmd, argv) {
+    setup() {
         // ghost setup --local, skip
-        if (argv.local) {
-            return;
-        }
+        const enabled = ({argv}) => !argv.local;
+
+        return [{
+            id: 'nginx',
+            name: 'Nginx',
+            task: (...args) => this.setupNginx(...args),
+            enabled
+        }, {
+            id: 'ssl',
+            name: 'SSL',
+            enabled,
+            task: (...args) => this.setupSSL(...args),
+            skip: ({tasks, ui}) => {
+                if (tasks.nginx.isSkipped()) {
+                    ui.log('Nginx setup task was skipped, skipping SSL setup', 'yellow');
+                    return true;
+                }
+
+                if (tasks.nginx.hasFailed()) {
```

Comment:
these two status checks could maybe be combined, seemed more prudent to leave the messaging different per case though.